lib/pagseguro/subscription/response_serializer.rb

```diff
@@ -9,9 +9,51 @@ def initialize(xml)
 
       def serialize
         {}.tap do |data|
-          data[:code] = xml.at_css('directPreApproval > code').text
+          data[:code] = xml.css('directPreApproval > code').text
         end
       end
+
+      def serialize_from_search
+        {}.tap do |data|
+          data[:name] = xml.at_css('name').text
+          data[:code] = xml.css('code').text
+          data[:date] = xml.css('date').text
+          data[:tracker] = xml.css('tracker').text
+          data[:status] = xml.css('status').text
+          data[:reference] = xml.css('reference').text
+          data[:last_event_date] = xml.css('lastEventDate').text
+          data[:charge] = xml.css('charge').text
+          data[:sender] = serialize_sender if xml.at_css('sender')
+        end
+      end
+
+      private
+
+      def serialize_sender(data={})
+        data[:name] = xml.at_css('sender > name').text
+        data[:email] = xml.at_css('sender > email').text
+        data[:phone] = serialize_phone if xml.at_css('phone')
+        data[:address] = serialize_address if xml.at_css('address')
+        data
```

Comment:
Actually here need to return a hash that will be set as a value of a key. My fault was to wait the *data* argument cause I don't use it.
For example:
```